code/__DEFINES/obj_flags.dm

```diff
@@ -10,7 +10,10 @@
 #define UNIQUE_RENAME			(1<<6) // can you customize the description/name of the thing?
 #define USES_TGUI				(1<<7)	//put on things that use tgui on ui_interact instead of custom/old UI.
 #define FROZEN					(1<<8)
-#define BLOCK_Z_FALL			(1<<9) // Should this object block z falling?
+#define BLOCK_Z_OUT_DOWN		(1<<9)  // Should this object block z falling from loc?
+#define BLOCK_Z_OUT_UP			(1<<10) // Should this object block z uprise from loc?
+#define BLOCK_Z_IN_DOWN			(1<<11) // Should this object block z falling from above?
+#define BLOCK_Z_IN_UP			(1<<12) // Should this object block z uprise from below?
```

Comment:
This might warrant its own atom-level variable.
Something like `z_density`
Either that with several flags, or two variables such as `z_floor_density` and `z_ceiling_density`

Reply:
Yes, great idea, but while we have place for this flags i prefer keep it as it is.
I preffer minimize amount of different vars.
Or we wait another montch when me(or someone) remake it.

Reply:
This must use CanAllowThrough

Reply:
All good then.